api/app/signals/apps/api/v1/views/signal.py

```diff
@@ -54,6 +55,17 @@ class PrivateSignalViewSet(mixins.CreateModelMixin, mixins.UpdateModelMixin, Dat
         'notes',
     ).all()
 
+    # Geography queryset to reduce the complexity of the query
+    geography_queryset = Signal.objects.select_related(
+        'location',  # We only need the location for now
+    ).filter(
+        location__isnull=False  # We can only show signals on a map that have a location
+    ).only(  # No need to select anything else than the id, created and the location for now
+        'id',
+        'created_at',
+        'location'
+    ).all()
```

Comment:
@tcoenen  I added a separate queryset for the geography query. This reduces a bit of the complexity. Also the filters still can be used